frontend/src/components/NodePathHistory/NodePathHistory.jsx

```diff
@@ -0,0 +1,167 @@
+import React, { useState, useEffect } from "react";
+// import styles from "./NodePathHistory.module.css";
+import Box from "@material-ui/core/Box";
+import Button from "@material-ui/core/Button";
+import IconButton from "@material-ui/core/IconButton";
+import Breadcrumbs from "@material-ui/core/Breadcrumbs";
+import TextField from "@material-ui/core/TextField";
+import ClickAwayListener from "@material-ui/core/ClickAwayListener";
+import Snackbar from "@material-ui/core/Snackbar";
+import Alert from "@material-ui/lab/Alert";
+import NavigateNext from "@material-ui/icons/NavigateNext";
+import EditRounded from "@material-ui/icons/EditRounded";
+import DoneRounded from "@material-ui/icons/DoneRounded";
+
+const MODES = { VIEW: 0, EDIT: 1 };
+
+/**
+ * Join the list of node names into one string
+ */
+function convertPathToString(path) {
+  return path.join("/");
+}
+
+export const NodePathHistory = ({
+  path,
+  onNodeNameClick,
+  onPathChange,
+  isValidPath,
+}) => {
+  const [mode, setMode] = useState(MODES.VIEW);
+  const [currentText, setCurrentText] = useState("");
+  const [errorOpen, setErrorOpen] = React.useState(false);
+
+  useEffect(() => {
+    setCurrentText(convertPathToString(path));
+  }, [path]);
+
+  /**
+   * Store the current text being typed
+   */
+  const onTextChangeHandler = (event) => {
+    setCurrentText(event.target.value);
+  };
+
+  /**
+   * Triggered when the user is finished editing the path
+   * and clicks done. Notifies the parent of the change
+   * through a callback, passing in the new path as a parameter.
+   */
+  const onDoneClickHandler = () => {
+    // Remove preceding or trailing slashes and spaces
+    const newPath = currentText
+      .trim()
+      .replace(/^\/+/, "")
+      .replace(/\/+$/, "")
+      .split("/");
+
+    // Inform the parent container of the change, if there was one and
+    // given that the path is valid
+    if (currentText !== convertPathToString(path)) {
+      if (isValidPath(newPath)) {
+        onPathChange(newPath);
+      } else {
+        setErrorOpen(true);
```

Comment:
When using this feature I observed there was a green checkmark beside the edit box; could we make that change between a :heavy_check_mark: and an :x: as the user types so that they know whether their path is valid or not before they hit enter?

Reply:
Good idea!

Reply:
Actually, I think it's better if we just disable the button if input is invalid instead of putting ❌. I pushed the changes. Check it out and let me know what you think.

Reply:
Looking good; just one thing could we make the path entry be not case sensitive? e.g. root/sass should still be correct even though the correct names are Root/SaaS

Reply:
Good point, I'll add that.

Reply:
Done.